### PR TITLE
Updated Italian localization

### PR DESCRIPTION
**Pull request by badinl**
Updated missing strings, in addition to various corrections and general improvements/clarifications. Some of the strings still sound a bit "off", but this should represent a 99% complete localization.

### CODE REVIEW

**Comment by badinl** [comment]
@selvanair, I have checked the localization file against the English version and all strings are now up to date. I have added new strings, updated a few (including eg. copyright headers) and corrected a few more inconsistencies.

I have also added keyboard shortcut support ("&" tokens) where missing.

File: res/openvpn-gui-res-it.rc [hunk 1]
```diff
@@ -17,7 +18,7 @@
  *  along with this program (see the file COPYING included with this
  *  distribution); if not, write to the Free Software Foundation, Inc.,
  *  59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
- */
+*/
```
Comment (selvanair) [inline res/openvpn-gui-res-it.rc]: Original looks right -- space removed by mistake?

**Comment by mattock** [comment]
I went through this PR already. While I'm not a native italian speaker, my laptops and mobiles have "spoken" italian for about 13 years and on top of that I use (mostly read) italian on a daily basis. The only thing that stood out from this translation I mentioned in my earlier comment already.

**Comment by selvanair** [comment]
@mattock: I somehow cannot see your previous comments on this.. Lost by force-push? Anyway, if it looks good, please merge. Thanks.

File: res/openvpn-gui-res-it.rc [hunk 2]
```diff
@@ -2,6 +2,7 @@
  *  OpenVPN-GUI -- A Windows GUI for OpenVPN.
  *
  *  Copyright (C) 2004 Mathias Sundman <mathias@nilings.se>
+ *                2009 Heiko Hund <heikoh@users.sf.net>
```
Comment (ordex) [inline res/openvpn-gui-res-it.rc]: why is this here? I am not sure how it is realated to the translation

File: res/openvpn-gui-res-it.rc (hunk 1 above)
Comment (badinl) [reply res/openvpn-gui-res-it.rc]: That's probably it. I'll just restore it.

File: res/openvpn-gui-res-it.rc (hunk 2 above)
Comment (badinl) [reply res/openvpn-gui-res-it.rc]: > 
> 
> @selvanair, I have checked the localization file against the English version and all strings are now up to date. I have added new strings, updated a few (including eg. copyright headers) and corrected a few more inconsistencies.
> 
> I have also added keyboard shortcut support ("&" tokens) where missing.

Please refer to the November 22 and 23 entries. As suggested by @selvanair, after my first PR, I checked the whole localization file against the English version and completed the job with the new fields. That included updating the copyright headers to match the ones on the English version.

Comment (selvanair) [reply res/openvpn-gui-res-it.rc]: But copyright indicates authorship of the file -- no point in adding someone else's name in a file you are editing :) I would leave it out though we have not been very strict/correct (IMO) about copyright year updates and attribution.. What I do is add/update my name and year when a substantial contribution is added to a file and never add or change anyone else's name or year.

Comment (badinl) [reply res/openvpn-gui-res-it.rc]: My mistake, I thought it was a project-wise copyright notice. I will revert this update.

File: res/openvpn-gui-res-it.rc [hunk 3]
```diff
@@ -371,34 +402,40 @@ BEGIN
...
     IDS_ERR_QUERY_SERVICE "Errore nell'interrogazione dello status del servizio."
-    IDS_ERR_SERVICE_START_FAILED "Fallito l'avvio di OpenVPN Service."
+    IDS_ERR_SERVICE_START_FAILED "Errore nell'avvio del servizio OpenVPN"
     IDS_ERR_OPEN_SCMGR "OpenSCManager fallito (%d)"
-    IDS_ERR_STOP_SERVICE "Fallito l'arresto di OpenVPN Service"
-    IDS_NFO_RESTARTED "OpenVPN Service riavviato."
+    IDS_ERR_STOP_SERVICE "Errore nell'arresto del servizio OpenVPN"
+    IDS_NFO_RESTARTED "Servizio OpenVPN riavviato."
+    IDS_ERR_ACCESS_SERVICE_PIPE "Accesso alla pipe del servizio fallito."
+    IDS_ERR_WRITE_SERVICE_PIPE "Scrittura alla pipe del servizio fallita."
```
Comment (ordex) [inline res/openvpn-gui-res-it.rc]: What do you think about "sulla pipe" instead of "alla pipe" ? But I am not even sure here. Normally I'd say "scrivi sul socket", so I thought about the same for a pipe.

File: res/openvpn-gui-res-it.rc [hunk 4]
```diff
@@ -411,8 +448,30 @@ BEGIN
...
     IDS_ERR_CREATE_REG_KEY "Errore nella creazione della chiave HKLM\\SOFTWARE\\OpenVPN-GUI."
-    IDS_ERR_OPEN_WRITE_REG "Fallita l'apertura del registro in scrittura. Puoi far partire questa applicazione " \
+    IDS_ERR_OPEN_WRITE_REG "Errore nell'apertura del registro in scrittura. Puoi far partire questa applicazione " \
                            "una volta che l'amministratore ha aggiornato il registro."
-    IDS_ERR_READ_SET_KEY "Errore nella lettura e settaggio della chiave di registro ""%s""."
+    IDS_ERR_READ_SET_KEY "Errore nella lettura e modifica della chiave di registro ""%s""."
     IDS_ERR_WRITE_REGVALUE "Errore nella scrittura del valore di registro ""HKEY_CURRENT_USER\\%s\\%s""."
+    
+    /* importation */
+    IDS_ERR_IMPORT_EXISTS "Esiste già una configurazione con il nome ""%s"" already exists."
```
Comment (ordex) [inline res/openvpn-gui-res-it.rc]: I guess you wanted to remove "already exists" at the end :)

File: res/openvpn-gui-res-it.rc [hunk 5]
```diff
@@ -411,8 +448,30 @@ BEGIN
...
+    IDS_ERR_IMPORT_FAILED "Errore nell'importazione del file. Non ho potuto creare il percorso seguente.\n\n" \
```
Comment (ordex) [inline res/openvpn-gui-res-it.rc]: How about using the colon ":" after "seguente"?

File: res/openvpn-gui-res-it.rc [hunk 6]
```diff
@@ -411,8 +448,30 @@ BEGIN
...
+    /* Token password related */
+    IDS_NFO_TOKEN_PASSWORD_CAPTION "OpenVPN - Password Token"
+    IDS_NFO_TOKEN_PASSWORD_REQUEST "Inserisci la Password/PIN per il token '%S'"
+
+    IDS_NFO_AUTH_PASS_RETRY "Credenziali errate. Riprova..."
+    IDS_NFO_KEY_PASS_RETRY  "Password errata. Riprova..."
+    IDS_ERR_INVALID_PASSWORD_INPUT "Carattere non valido nella password"
+    IDS_ERR_INVALID_USERNAME_INPUT "Carattere non valido nel nome utente"
+    IDS_NFO_AUTO_CONNECT    "Connessione automatica tra %u secondi..."
+    IDS_NFO_CLICK_HERE_TO_START "L'interfaccia OpenVPN è già in esecuzione. Clicca con il tasto destro sull'icona della tray per cominciare."
```
Comment (ordex) [inline res/openvpn-gui-res-it.rc]: instead of "icona della tray", how about "icona sulla barra delle applicazioni"? I believe (but I am not sure, so please correct me) that this is how that area is called in Italian (it may have changed with the latest windows releases). O "nell'area di notifica", but I think the first is more understandable.

**Comment by ordex** [comment]
I am not the maintainer of this repository, but normally I always suggest to modify original commits rather than adding new ones. This will keep the git history clean. my 2 cents.

**Comment by cron2** [comment]
Hi,

On Mon, Feb 11, 2019 at 03:10:52PM -0800, Antonio Quartulli wrote:
> I am not the maintainer of this repository, but normally I always suggest to modify original commits rather than adding new ones. This will keep the git history clean. my 2 cents.

Yes, having this squashed into a single commit would be best.

gert

-- 
"If was one thing all people took for granted, was conviction that if you 
 feed honest figures into a computer, honest figures come out. Never doubted 
 it myself till I met a computer with a sense of humor."
                             Robert A. Heinlein, The Moon is a Harsh Mistress

Gert Doering - Munich, Germany                             gert@greenie.muc.de

**Comment by mattock** [comment]
Hmm, this seems to happen in all my GitHub projects when I add inline comments to code. I need to debug this _now_ it seems. Anyways, my only comment was about this English section:

```
    GROUPBOX "User Interface", 201, 6, 12, 235, 30
    LTEXT "&Language:", ID_TXT_LANGUAGE, 17, 25, 52, 12
```

I'm not sure of the context, but "Spazio di lavoro" seems odd. If I translate it back to english it becomes "Workspace", which is not the same as "User interface". For example, on Linux "Spazio di lavoro" is generally used to describe multiple workspaces. Would a more direct translation of "User interface", such as "Interfaccia utente" work here in your opinion?

Good thing that @ordex had more comments. I'd make some comments of my own if you guys could see them :smile:.

**Comment by ordex** [comment]
> Hmm, this seems to happen in all my GitHub projects when I add inline comments to code. I need to debug this _now_ it seems. Anyways, my only comment was about this English section:
> 
> ```
>     GROUPBOX "User Interface", 201, 6, 12, 235, 30
>     LTEXT "&Language:", ID_TXT_LANGUAGE, 17, 25, 52, 12
> ```
> 
> I'm not sure of the context, but "Spazio di lavoro" seems odd. If I translate it back to english it becomes "Workspace", which is not the same as "User interface". For example, on Linux "Spazio di lavoro" is generally used to describe multiple workspaces. Would a more direct translation of "User interface", such as "Interfaccia utente" work here in your opinion?

I agree with Samuli here. I hadn't checked the original English wording in this case.
If the starting point is "User interface" I think it should definitely be translated to "Interfaccia utente".

File: res/openvpn-gui-res-it.rc [hunk 7]
```diff
@@ -42,48 +43,48 @@ EXSTYLE WS_EX_TOPMOST
...
+    PUSHBUTTON "&OK", IDOK, 20, 92, 50, 14, BS_PUSHBUTTON | WS_TABSTOP | WS_DISABLED
+    PUSHBUTTON "&Annulla", IDCANCEL, 90, 92, 52, 14
     LTEXT "", ID_TXT_WARNING, 6, 108, 150, 10
 END
 
 /* Challenge Response Dialog */
 ID_DLG_CHALLENGE_RESPONSE DIALOG 6, 18, 212, 72
 STYLE WS_SIZEBOX | WS_POPUP | WS_VISIBLE | WS_CAPTION | DS_CENTER | DS_SETFOREGROUND
-CAPTION "OpenVPN - Challenge Response"
+CAPTION "OpenVPN - Risposta alla richiesta"
```
[the remark this answers is not in the document]
Comment (badinl) [reply res/openvpn-gui-res-it.rc]: Makes a lot of sense. I'll revert that.

File: res/openvpn-gui-res-it.rc [hunk 8]
```diff
@@ -149,47 +150,48 @@ CAPTION "Generale"
...
-    GROUPBOX "Preferences", 202, 6, 82, 235, 90
-    AUTOCHECKBOX "Append to log", ID_CHK_LOG_APPEND, 17, 95, 60, 10
-    AUTOCHECKBOX "Show script window", ID_CHK_SHOW_SCRIPT_WIN, 17, 110, 200, 10
-    AUTOCHECKBOX "Silent connection", ID_CHK_SILENT, 17, 125, 200, 10
-    LTEXT "Show Balloon", ID_TXT_BALLOON, 17, 140, 100, 10
-    AUTORADIOBUTTON "On connect", ID_RB_BALLOON1, 28, 155, 50, 10, WS_GROUP | WS_TABSTOP
-    AUTORADIOBUTTON "On connect/reconnect", ID_RB_BALLOON2, 86, 155, 90, 10
-    AUTORADIOBUTTON "Never", ID_RB_BALLOON0, 181, 155, 40, 10
+    GROUPBOX "Preferenze", 202, 6, 82, 235, 90
+    AUTOCHECKBOX "Aggiungi al f&ile di log", ID_CHK_LOG_APPEND, 17, 95, 60, 10
```
[the remark this answers is not in the document]
Comment (badinl) [reply res/openvpn-gui-res-it.rc]: It does, and IIRC I did ponder that specific wording, but decided that I couldn't find any real ambiguity in the simplified version, given that we're talking about a plaintext log file. I'm fine with changing it though.

File: res/openvpn-gui-res-it.rc [hunk 9]
```diff
@@ -236,37 +239,52 @@ BEGIN
...
+                        "Usa il menu ""Importa file.."" o copia i tuoi file in ""%s"" oppure ""%s""."
+    IDS_ERR_CONFIG_NOT_AUTHORIZED "L'avvio di questa connessione (%s) richiede di essere membro del "\
+                                  "gruppo ""%s"". Contatta il tuo amministratore di sistema.\n"
+    IDS_ERR_CONFIG_TRY_AUTHORIZE  "L'avvio di questa connessione (%s) richiede di essere membro del "\
+                                  "gruppo ""%s"".\n\n"\
+                                  "Vuoi aggiungerti al gruppo?\n"\
+                                  "Quest'azione può richiedere privilegi amministrativi."
+    IDS_NFO_CONFIG_AUTH_PENDING   "L'avvio di questa connessione (%s) richiede di essere membro del "\
+                                  "gruppo ""%s"".\n\n"\
+                                  "Completa il dialogo di autorizzazione precedente."
```
[the remark this answers is not in the document]
Comment (badinl) [reply res/openvpn-gui-res-it.rc]: I probably meant it as a shortened version of "finestra di dialogo". Indeed, this was a new translation from the English version "Please complete the previous authorization dialog.".

It does sound iffy, though. I'll change it to something better.

File: res/openvpn-gui-res-it.rc [hunk 10]
```diff
@@ -306,57 +324,70 @@ BEGIN
...
-                  "--show_script_window\t: 0=Nascondi la finestra di esecuzione dello script, 1=Mostrala.\n" \
-                  "--passphrase_attempts\t: Numero di tentativi concessi per la frase di sicurezza.\n" \
-                  "--connectscript_timeout\t: Tempo d'attesa affinché lo script di connessione finisca.\n" \
-                  "--disconnectscript_timeout\t: Tempo d'attesa affinché lo script di disconnessione finisca.\n" \
-                  "--preconnectscript_timeout\t: Tempo d'attesa affinché lo script di preconnessione finisca.\n"
+                  "--config_dir\t\t: Percorso della cartella dei file di configurazione.\n" \
+                  "--ext_string\t\t: Estensione dei file di configurazione.\n" \
+                  "--log_dir\t\t\t: Percorso della cartella dei file di log.\n" \
+                  "--priority_string\t\t: Stringa di priorità (Vedi install.txt per maggiori informazioni).\n" \
+                  "--append_string\t\t: 1=Aggiungi in coda al log. 0=Tronca log alla connessione.\n" \
```
[the remark this answers is not in the document]
Comment (badinl) [reply res/openvpn-gui-res-it.rc]: This is - I suppose - one of those cases where I think most users would refer to the operation with its English name, just like "append" - only "aggiungi in coda" is a commonly used translation, and I'm not sure if "truncate" has a similar one. I'm 50-50 on this.

File: res/openvpn-gui-res-it.rc (hunk 4 above)
Comment (badinl) [reply res/openvpn-gui-res-it.rc]: Ouch.

File: res/openvpn-gui-res-it.rc (hunk 5 above)
Comment (badinl) [reply res/openvpn-gui-res-it.rc]: Sounds good, but the English version would have the same problem then.

File: res/openvpn-gui-res-it.rc (hunk 6 above)
Comment (badinl) [reply res/openvpn-gui-res-it.rc]: Yeah, that would be "area di notifica". I'll change it.

File: res/openvpn-gui-res-it.rc (hunk 3 above)
Comment (badinl) [reply res/openvpn-gui-res-it.rc]: I'll change the row 434 occurrence as your version does sound better for that operation.

**Comment by badinl** [comment]
> I'm not sure of the context, but "Spazio di lavoro" seems odd. If I translate it back to english it becomes "Workspace", which is not the same as "User interface". For example, on Linux "Spazio di lavoro" is generally used to describe multiple workspaces. Would a more direct translation of "User interface", such as "Interfaccia utente" work here in your opinion?

Thanks for noticing that error, @mattock, when I installed OpenVPN that was actually the one string that made me want to see if I could help with the translation. For the record, the previous Italian version was "Superficie di lavoro", which didn't make much sense, so I had changed that without looking at what it was meant to be translated to.

**Comment by badinl** [comment]
Alright, I've applied the suggested changes and squashed the commits. (I could really have picked a better commit name for that though)

**Comment by selvanair** [comment]
Please do give an appropriate descriptive name like "Updated Italian localizaton" to the commit. And preferably add a sign-off (git commit -s). Thanks.

**Comment by badinl** [comment]
Done. Sorry guys, I really don't use Git as often as I'd like.

File: res/openvpn-gui-res-it.rc [hunk 11]
```diff
@@ -42,48 +42,48 @@ EXSTYLE WS_EX_TOPMOST
...
+    PUSHBUTTON "&OK", IDOK, 20, 92, 50, 14, BS_PUSHBUTTON | WS_TABSTOP | WS_DISABLED
+    PUSHBUTTON "&Annulla", IDCANCEL, 90, 92, 52, 14
     LTEXT "", ID_TXT_WARNING, 6, 108, 150, 10
 END
 
 /* Challenge Response Dialog */
 ID_DLG_CHALLENGE_RESPONSE DIALOG 6, 18, 212, 72
 STYLE WS_SIZEBOX | WS_POPUP | WS_VISIBLE | WS_CAPTION | DS_CENTER | DS_SETFOREGROUND
-CAPTION "OpenVPN - Challenge Response"
+CAPTION "OpenVPN - Challenge response"
```
Comment (ordex) [inline res/openvpn-gui-res-it.rc]: shouldn't this just remain the same as it was before: "**R**esponse" ?

**Comment by ordex** [comment]
Other than my last minor comment, the PR looks good. Thanks @badinl for your work!

**Comment by badinl** [comment]
My pleasure. Pushed the CR correction w/signoff and squashed commits. All should be good now.

**Comment by selvanair** [comment]
The only change is response -> Response that @ordex pointed out. Merging. 
Thanks.
